src/framework/theme/components/popover/popover.directive.ts

```diff
@@ -311,25 +319,40 @@ export class NbPopoverDirective implements OnInit, OnDestroy {
       return this.calcAdjustment(placed, host);
     }
 
-    return this.calcPosition(placed, host);
+    return {
+      position: this.calcPosition(placed, host),
+      placement: this.physicalPlacement,
+    };
+  }
+
+  /*
+   * Maps logical position to physical according to current layout direction.
+   * */
+  private get physicalPlacement(): NbPopoverPlacement {
```

Comment:
I think it's quite complicated solution for rtl support. Maybe we can add *START* and *END* in *NbPopoverPlacement* instead of creating separate enum. And then extend *NbPositioningHelper* with calculators which can choose between *left* and *right* calculators based on the *NbLayoutDirectionService*.
I think it'll simply and will require fewer changes.
What do you think about this solution?